Jetpack Settings: Handle updates action in the settings reducer

oskosk opened the pull request:
Updates the `state/jetpack-settings/settings` reducer to handle this action.

#### Testing instructions

1. Run `npm run test-client client/state/jetpack-settings/settings/``
2. Expect the tests to pass


#### Why 

We were not reducing this action, thus updating a setting through the action creators didn't have any effect on the state after the API request was successful.

oskosk commented on client/state/jetpack-settings/settings/test/reducer.js:
```diff
@@ -84,9 +84,9 @@ describe( 'reducer', () => {
 					12345678: SETTINGS_FIXTURE[ siteId ]
 				},
 				action = {
-					type: JETPACK_SETTINGS_RECEIVE,
```
@tyxla is this change here doing the right thing? Looks to me that this test was thought for this case, but referencing the wrong action type, but I'm not 100% sure

tyxla replied:
The test for `JETPACK_SETTINGS_RECEIVE` should still remain; I suggest introducing an additional similar test for `JETPACK_SETTINGS_UPDATE_SUCCESS`, keeping the existing test for `JETPACK_SETTINGS_RECEIVE`.

You may ask why? Well, because the reducer should be updating the settings in the 2 cases:

* After fetching the settings
* After updating the settings (which also returns the updated settings)

So let's keep these 2 cases, and add appropriate tests for both of them.

oskosk replied:
Looks like there was already a test for _RECEIVE, and the one I updated was a duplicated one. Is it possible ?

oskosk replied:
Either that or I totally misparsed that test's description should update settings in the items object

tyxla replied:
I'm not seeing a duplicated test. The one you've updated is a valid test that I believe should remain as is.

oskosk commented:
@tyxla I've updated this PR like this:

[x] Add a `createItemsReducer()` factory for handling items
[x] Add a test for reducer's behasviour when JETPACK_SETTINGS_UPDATE_SUCCESS.
[x] Replaced the JETPACK_SETTINGS_RECEIVE test description which stated `should update settings in the items object` for `should replace settings in the items object when settings are already loaded for a site`
